Spring-Project/src/main/java/com/epam/project/spring/repository/impl/UserRepositoryImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.epam.project.spring.repository.impl;
+
+import com.epam.project.spring.model.User;
+import com.epam.project.spring.repository.UserRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class UserRepositoryImpl implements UserRepository {
+
+    private final List<User> listUser = new ArrayList<>();
+
+    @Override
+    public User getUser(String email) {
+        return listUser.stream().filter(user -> user.getEmail().equals(email)).findFirst().orElseThrow(() -> new RuntimeException("User not found"));
+    }
+
+    @Override
+    public List<User> listUsers() {
+        return listUser;
+    }
+
+    @Override
+    public User createUser(User user) {
+         listUser.add(user);
```

Comment:
Hi Zenyk. Since we are using '**email**' field as identifier, please implement additional check to verify whether **User** with such email already exists. This will allow avoid **User** duplication. In case if user with the same email exists please throw custom Exception with neutral message:
"_Error occurred during user creation. Please try again_"
Thank you